chiron/reporters.py

```diff
@@ -62,51 +123,222 @@ def _write_to_disk(self, key):
-            log.debug(f"Creating {key} in {self.filename}")
-            self.h5file.create_dataset(
-                key, data=data, maxshape=(None,) + data.shape[1:], chunks=True
+        log.debug(f"Writing {key} to file")
+        if key == "positions" and hasattr(self, "_write_to_trajectory"):
+            xyz = np.stack(self.buffer[key])
+            self._write_to_trajectory(
+                positions=xyz,
             )
 
-        self.buffer[key] = []
+        with h5py.File(self.log_file_path, "a") as h5file:
+            if key in h5file:
+                data = np.array(self.buffer[key])
+                dset = h5file[key]
+                dset.resize((dset.shape[0] + data.shape[0],) + data.shape[1:])
+                dset[-data.shape[0] :] = data
+            else:
+                data = np.array(self.buffer[key])
+                log.debug(f"Creating {key} in {self.log_file_path}")
+                h5file.create_dataset(
+                    key, data=data, maxshape=(None,) + data.shape[1:], chunks=True
+                )
+
+    def reset_reporter_file(self):
+        # delete the reporter files
+        import os
+
+        # if file exists, delete it
+        if os.path.exists(self.log_file_path):
+            log.debug(f"Deleting {self.log_file_path}")
+            os.remove(self.log_file_path)
 
-    def close(self):
+    def flush_buffer(self) -> None:
         """
-        Write any remaining data in the buffer to disk and close the HDF5 file.
+        Write any remaining data in the buffer to disk.
 
         """
         for key in self.buffer:
             if self.buffer[key]:
                 self._write_to_disk(key)
-        self.h5file.close()
+        self._reset_buffer()
 
-    def get_property(self, name: str):
+    def _reset_buffer(self) -> None:
         """
-        Get the property from the HDF5 file.
+        Reset the data buffer after writing to disk.
+        """
+        self.buffer = {key: [] for key in self.buffer}
+
+    def get_property(self, name: str) -> np.ndarray:
+        """
+        Retrieve a specific property from the HDF5 file.
 
         Parameters
         ----------
         name : str
-            Name of the property to get.
+            The name of the property to retrieve.
 
         Returns
         -------
         np.ndarray
-            The property.
+            The retrieved property data, if available.
+        """
+        if name == "positions" and hasattr(self, "read_from_trajectory"):
+            return self.read_from_trajectory()
+
+        with h5py.File(self.log_file_path, "r") as h5file:
+            if name in h5file:
+                data = np.array(h5file[name])
+            elif name in self.buffer and name not in h5file:
+                data = np.array(self.buffer[name])
+            elif name not in h5file:
+                log.warning(f"{name} not in HDF5 file")
+                return None
+
+            if name == "u_kn":
+                return np.transpose(
```

Comment:
I think it may be better to store this not as the transposed array...that is, keep it, n_interations, n_state, n_replicas, as that is how we encode the other pieces of data we are logging. 

The reason to transpose is for mbar evaluation; it seems like maybe it makes more sense to do the transpose before calling mbar.

Reply:
We discussed this today and I think we are in agreement on storing with the same convention as other data, and transpose either in the call to the free energy calculation or before we pass to the free energy calculation.

Reply:
Fixed, so we transpose when we call mbar